sites/main-site/src/content/blog/2024/seqera-containers-part-2.mdx

```diff
@@ -0,0 +1,639 @@
+
+As [described above](#pipelines), pipelines will have a set of config files automatically generated
+that specify the container or conda environment for each process in the pipeline.
+
+Creation of these files will be triggered whenever installing, updating or removing a module, via the `nf-core` CLI.
+The config files will be completely regenerated each time, so there will never be any manual merging required.
+
+The trickiest part of this process is linking the module containers to the pipeline processes.
+Modules can be imported into pipelines with any alias or scope.
+We need to match this against the values that we find in the module `meta.yml` files:
+
+-   Run `nextflow inspect` to generate default Docker `linux/arch64` config
+-   Copy this file and replace the container names with the relevant container for each platform,
+    using the `meta.yml` files that match the Docker container name
+
+This is why we duplicate the default docker container in both `main.nf` and `meta.yml` files -
+it allows us to link the module to the pipeline.
+
+:::warning
+
+Currently, `nextflow inspect` works by executing a dry-run of the pipeline.
+We can run using `-profile test`, but any processes that are skipped due to pipeline logic
+will not be included in the config file.
+
+This is a known limitation and we are currently working hard on a new version of `nextflow inspect`
+which will return all processes, regardless of whether they are skipped or not.
+This will be a requirement for progression with the Seqera Containers migration.
+
+:::
+
+:::note
+
+This process of copying files and using string substituion is a bit of a hack.
+If you have any ideas on how to improve this, please let us know!
+
+:::
+
+## Edge cases and local modules
+
+There will always be edge cases that won't fit the automation described above.
+Not all software tools can be packaged on Bioconda (though we encourage it where possible!).
+For example, some tools have licensing restrictions that prevent them from being distributed in this way.
+Other edge-cases include optional usage of container variants for GPU support, or other hardware.
+
+Local modules won't be able to benefit from the Wave CLI automation to fetch containers from Seqera Containers
+and will have to be manually updated by the pipeline developer.
+
+For these reasons, we will still support custom `container` declarations in modules
+without use of Seqera Containers. It will be up to the module contributors to ensure that these
+are correctly specified and kept up to date manually.
+
+These can be specified in the `main.nf` file and added to the autogenerated platform-specific config files,
+as long as they remain above the comment line:
+
+```groovy
+// AUTOGENERATED CONFIG BELOW THIS POINT - DO NOT EDIT
+```
+
+If at all possible then software should be packaged with Bioconda and Seqera Containers.
+Failing that, custom containers should be stored under the nf-core account on the quay.io Docker registry.
```

Comment:
```suggestion
Failing that, custom containers should be stored under the nf-core account on the quay.io Docker registry.
[Custom containers can be built with wave in CI, and pushed to quay.io automatically](https://github.com/nf-core/modules/pull/4940) using the same mechanisms as the `environment.yml` files and replacing it with the `Dockerfile`s.
```

Reply:
Applied with a few tweaks in 4faacdd6c 👍🏻